### PR TITLE
fix: [lw-11830]: handle not delegated stake keys scenarios in send flow

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] JIRA - [LW-11830](https://input-output.atlassian.net/browse/LW-11830)
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Screenshot 2024-11-27 at 15 19 35

[LW-11830]: https://input-output.atlassian.net/browse/LW-11830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**processReports**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/7428/11912060633/index.html) for [18975ed6](https://github.com/input-output-hk/lace/pull/1543/commits/18975ed6ce4766cab073ab650a8c026d4b92e50d)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 31     | 1      | 4       | 0     | 36    | ❌     |

File: apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/Form/CoinInput/CoinInput.tsx [hunk 1]
```diff
@@ -20,24 +29,57 @@ export const CoinInput = ({
...
-    />
+    <>
+      <AssetInputList
+        disabled={(!!assetBalances && assetBalances?.size === 0) || !canAddMoreAssets}
+        rows={selectedCoins}
+        onAddAsset={onAddAsset}
+        translations={{ addAsset: t('browserView.transaction.send.advanced.asset') }}
+        isPopupView={isPopupView}
+      />
+      {isPopupView ? (
```
Comment (mirceahasegan) [inline apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/Form/CoinInput/CoinInput.tsx:62]: I do not understand this part. Are we always going to display the CTA?

File: apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/Form/Form.tsx [hunk 2]
```diff
@@ -75,25 +78,29 @@ export const Form = ({
...
-        if (isInMaxAmountList && !isInInsufficientBalanceList) {
-          return [...prevInputsList, lastFocusedInput]; // add it to the insufficient balance list
-          // check if the last focused element has balance and exists in insufficient balance list
-        } else if (!isInMaxAmountList && isInInsufficientBalanceList) {
-          return prevInputsList.filter((inputId) => inputId.split('.')[1] !== id); // remove all items with same coin id (cardano id or asset id)
-        }
-
-        return prevInputsList;
-      });
+      setInsufficientBalanceInputs(getNextInsuffisientBalanceInputs(lastFocusedInput, reachedMaxAmountList, id));
```
Comment (mirceahasegan) [inline apps/browser-extension-wallet/src/views/browser-view/features/send-transaction/components/Form/Form.tsx:93]: typo nitpick:
```diff
-      setInsufficientBalanceInputs(getNextInsuffisientBalanceInputs(lastFocusedInput, reachedMaxAmountList, id));
+      setInsufficientBalanceInputs(getNextInsufficientBalanceInputs(lastFocusedInput, reachedMaxAmountList, id));
```

File: apps/browser-extension-wallet/src/views/browser-view/features/staking/hooks.ts [hunk 3]
```diff
@@ -17,3 +19,55 @@ export const useDelegationTransaction = (): { signAndSubmitTransaction: () => Pr
 
   return { signAndSubmitTransaction };
 };
+
+export const useRewardAccountsData = (): {
+  accountsWithRegisteredStakeCredsWithoutVotingDelegation: Wallet.Cardano.RewardAccountInfo[];
+  accountsWithRegisteredStakeCreds: Wallet.Cardano.RewardAccountInfo[];
+  poolIdToRewardAccountMap: Map<string, Wallet.Cardano.RewardAccountInfo>;
```
Comment (mirceahasegan) [inline apps/browser-extension-wallet/src/views/browser-view/features/staking/hooks.ts:26]: Multiple rewardAccounts could be delegated to the same stake pool.
Should this be:
```diff
-  poolIdToRewardAccountMap: Map<string, Wallet.Cardano.RewardAccountInfo>;
+  poolIdToRewardAccountMap: Map<string, Wallet.Cardano.RewardAccountInfo[]>;
```

Comment (vetalcore) [reply apps/browser-extension-wallet/src/views/browser-view/features/staking/hooks.ts:26]: i've just copied this hook from https://github.com/input-output-hk/lace/pull/1537, so will resolve there and rebase once merged

**Review by mirceahasegan — APPROVED** [review]
Fantastic work @vetalcore !

File: apps/browser-extension-wallet/src/views/browser-view/features/staking/hooks.ts [hunk 4]
```diff
@@ -17,3 +19,55 @@ export const useDelegationTransaction = (): { signAndSubmitTransaction: () => Pr
...
+  lockedStakeRewards: BigInt;
+} => {
+  const { inMemoryWallet } = useWalletStore();
+  const rewardAccounts = useObservable(inMemoryWallet.delegation.rewardAccounts$);
+  const accountsWithRegisteredStakeCreds = rewardAccounts?.filter(
+    ({ credentialStatus }) => Wallet.Cardano.StakeCredentialStatus.Registered === credentialStatus
+  );
+
+  const accountsWithRegisteredStakeCredsWithoutVotingDelegation = useMemo(
+    () => accountsWithRegisteredStakeCreds?.filter(({ dRepDelegatee }) => !dRepDelegatee),
```
Comment (vetalcore) [inline apps/browser-extension-wallet/src/views/browser-view/features/staking/hooks.ts:36]:
```suggestion
    () => accountsWithRegisteredStakeCreds?.filter(({ dRepDelegatee }) => !dRepDelegatee || dRepDelegatee.delegateRepresentative.active),
```

**Pull request merged**